demo/Demo.hpp

```diff
@@ -0,0 +1,130 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+#ifndef PEGASUS_DEMO_HPP
+#define PEGASUS_DEMO_HPP
+
+#include <pegasus/ParticleWorld.hpp>
+#include <demo/Render.hpp>
+
+namespace pegasus
+{
+/**
+ * @brief Represents a singletone instance of the Demo
+ */
+class Demo
+{
+public:
+    struct Object;
+
+    /**
+     * @brief Returns a reference to the singletone Demo instance
+     * @return Demo object reference
+     */
+    static Demo& GetInstance();
+
+    /**
+     * @brief Checks if the window is not closed and render is initialized
+     * @return @c true if initialize, @c false otherwise
+     */
+    bool IsValid() const;
+
+    /**
+     * @brief Computes and renders frame
+     */
+    void RunFrame();
+
+    /**
+     * @brief Makes physical and render instance of the line and returns it
```

Comment:
`Create` would be a better word for this kind of thing.
Stop documenting `and returns it` - it is obvious from `@return`.
Also it looks like in implementation you actually do not create `physical instance of the line`
What does it mean to `Create physical and render instance`? Some light description would do the trick:

```cpp
/** @brief Creates an object describing a line
 *
 *  The object contains only render-related information and is not registered in physics world
 *
 *  @param [...]
 *
 *  @return a reference to newly created Object
 */
```